### PR TITLE
ENH: infer_objects copy kwd

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

File: pandas/core/internals/array_manager.py [hunk 1]
```diff
@@ -374,19 +374,22 @@ def fillna(self: T, value, limit, inplace: bool, downcast) -> T:
...
                     convert_datetime=True,
                     convert_timedelta=True,
                     convert_period=True,
                 )
+                if result is arr and copy:
+                    return arr.copy()
+                return result
             else:
-                return arr.copy()
+                return arr.copy() if copy else arr
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/core/internals/array_manager.py:392]: > is there a test that hits this?

TestInferObjects.test_copy should hit this in the CI build specific to ArrayManager

> the plan is to "https://github.com/pandas-dev/pandas/issues/48880#issuecomment-1279611971" anyway

I'd like this, but have no plans to push for it.

**Comment by jorisvandenbossche** [comment]
@jbrockmendel for 1.5 we had a discussion related to new `copy` keywords (https://github.com/pandas-dev/pandas/issues/48141) and ended up reverting some of those new `copy` keywords. Is there a specific reason to still add a `copy` keyword to `infer_objects`?

**Comment by jbrockmendel** [comment]
Two big motivations.  One was that we removed automatic inference in a few places, telling users to use infer_objects instead.  Wanted to allow that operation to be cheap where possible.  Second is that there was a bug in here in Block[Subclass].convert that was inconsistent about making copies.  Fixing that bug required making more copies, and it's worth softening that blow.

**Pull request merged**